tool/requote.js

```diff
@@ -396,7 +396,7 @@ function requoteSentence(id, context, sentence, program, mode, requote_numbers,
     let contextEntities = new Set;
     if (context) {
         for (let token of context.split(' ')) {
-            if (/^[A-Z]/.test(token))
+            if (/^[A-Z].*_[0-9|۰-۹]/.test(token)) // match strings like QUOTED_STRING_0, NUMBER_0 and GENERIC_ENTITY_uk.ac.cam.multiwoz.Hotel:Hotel_0
```

Comment:
The syntax is wrong (it will match the literal character `|` instead of matching either latin or persian digits). But why do you want persian digits here at all?

Reply:
Also, this is wrong if you have more than 10 entities of a certain type (which can happen in rare cases with dialogues)

Reply:
You are right. I copied from lines 34-36 of the same file, which are wrong too.

Reply:
Yeah but why persian numbers in entities? It doesn't make sense.

Reply:
Yeah, that part is removed in the `wip/mehrad/updates` branch. I removed it here as well.